src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -67,12 +90,31 @@ export function useRefetchQuoteCallback() {
           fee
         })
       } catch (error) {
-        console.error('Error getting the quote (price/fee)', error)
+        const errorType = error?.errorType
+        console.error('Error getting the quote (price/fee):', errorType)
+
+        // Unsupported token
+        if (errorType === ApiErrorCodes.UnsupportedToken) {
+          // TODO: will change with introduction of data prop in error responses
+          const unsupportedTokenAddress = error?.description?.split(' ')[2]
+
+          console.error(
+            '[useRefetchPriceCallback]::Unsupported token detected:',
+            unsupportedTokenAddress,
+            '- disabling.'
+          )
+
+          addUnsupportedToken({
+            chainId,
+            address: unsupportedTokenAddress.toLowerCase(),
+            dateAdded: Date.now()
+          })
+        }
```

Comment:
nitpicking, I would suggest to create a function, `handleUnsupportedToken` with the body of the if. You make the whole logic easier to read. Note that this hook has already some complexity callback/try/catch/if so it's nice to see to see something like ``handleUnsupportedToken({ error, chainId, addUnsupportedToken })` So we don't need to enter in the implementation if we don't want.